app/src/main/java/com/dantsu/thermalprinter/MainActivity.java

```diff
@@ -81,8 +102,18 @@ public void printBluetooth() {
         if (ContextCompat.checkSelfPermission(this, Manifest.permission.BLUETOOTH) != PackageManager.PERMISSION_GRANTED) {
             ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.BLUETOOTH}, MainActivity.PERMISSION_BLUETOOTH);
         } else {
-            // this.printIt(BluetoothPrintersConnections.selectFirstPaired());
-            new AsyncBluetoothEscPosPrint(this).execute(this.getAsyncEscPosPrinter(null));
+
+            //select first paired print
+//             this.printIt(BluetoothPrintersConnections.selectFirstPaired());
+            //select first paired print with async
+//            new AsyncBluetoothEscPosPrint(this).execute(this.getAsyncEscPosPrinter(null));
+            //select device for print
+            if(selectedDevice != null){
+                this.printIt(BluetoothPrintersConnections.selectDevice(selectedDevice));
```

Comment:
You do a synchronized print instead of asynchronized print. That's not the correct way.